llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -824,6 +825,11 @@ void AMDGPUTargetMachine::registerPassBuilderCallbacks(PassBuilder &PB) {
           PM.addPass(AMDGPULowerModuleLDSPass(*this));
         if (EnableAMDGPUAttributor && Level != OptimizationLevel::O0)
           PM.addPass(AMDGPUAttributorPass(*this));
+        if (!NoKernelInfoEndLTO) {
```

Comment:
I don't see the point of the option

Reply:
See [this comment](https://github.com/llvm/llvm-project/pull/102944#discussion_r1797494663) and [these docs](https://github.com/llvm/llvm-project/blob/d3beccfe9eb34636cd0015b867d6fcda8fa6ea26/llvm/docs/KernelInfo.rst?plain=1#L43-L63).